torchtune/datasets/_packed.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Dict, List, Optional, Tuple
+
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+
+class PackedDataset(Dataset):
```

Comment:
Also, maybe call this GreedyPackedDataset? For when we add different versions?

Reply:
mm I'm inclined to leave it as is and rename it when we add different versions if needed. don't want to jump the gun